source/simulator/solver.cc

```diff
@@ -714,8 +714,8 @@ namespace aspect
     SolverControl solver_control_cheap (parameters.n_cheap_stokes_solver_steps,
                                         solver_tolerance);
 
-    SolverControl solver_control_expensive (system_matrix.block(block_vel,block_p).m() +
-                                            system_matrix.block(block_p,block_vel).m(), solver_tolerance);
+    SolverControl solver_control_expensive (parameters.n_expensive_stokes_solver_steps,
```

Comment:
Hm -- in your case, you limit the number of iterations to 1000 by default, but it may have been much larger before. Does that matter? Can we think of cases where it may ever get to more than 1000 in normal operations?

In any case, I'd like to use
```
  std::min (sys...m() + sys...m(), parameters.n_expensive_stokes_solver_steps)
```
Nothing good can happen if we use more iterations than there are unknowns in the linear system.